tests/components/suez_water/conftest.py

```diff
@@ -37,39 +38,55 @@ def mock_setup_entry() -> Generator[AsyncMock]:
 
 
 @pytest.fixture(name="suez_client")
-def mock_suez_client() -> Generator[MagicMock]:
+def mock_suez_client() -> Generator[AsyncMock]:
     """Create mock for suez_water external api."""
     with (
+        patch("pysuez.async_client.SuezAsyncClient", autospec=True) as main,
```

Comment:
not needed?

Reply:
Yeah apparently.

I did add it when fixing the tests but I didn't understand why it wan't working without 😄 and now it's working without it.
So it was probably something else.
I'm still learning python and everything with it testing / mocking which is all very different from the languages I know

Reply:
Hehe, I understand, but that's also why I'm pushing back :)